sections/advanced/combined-code-coverage.md

```diff
@@ -0,0 +1,304 @@
+# Combined Unit & E2E Code Coverage: case study on a real life system using Angular, Jest, Cypress & GitLab
+
+<br/><br/>
+
+### One Paragraph Explainer
+
+There are a variety of metrics to evaluate to have high confidence in quality. Unit test coverage gives us high confidence at source code level. For example using Angular CLI, a placeholder spec file is auto-generated for a given component. In contrast E2E tests can cover an interaction of components; whether these are state transitions -think of state machines-, feature use cases, application workflows or any combination of using multiple components at a time. This is where they provide the most value. Once we have a decent amount of unit test coverage, we cover our cases with ui-e2e tests, with green pipelines we can have high confidence in quality.
+
+The web has evolved and now we can also gain source-code code coverage from ui-e2e tests, filling any gaps in source-code coverage through a guided process.
+
+<br/>
+
+## Foreword
+
+By now there are a plethora of examples and resources on combined code coverage. We wanted to study a CI agnostic, real life example with a particular stack, without reliance on external services, and explain the why behind the decisions that are being made.
+
+The real life system under test stack is **Angular, Jest, Cypress & GitLab**.
+
+> Jest can be replaced with Karma; all we need from unit tests is the default *coverage* folder to be generated.
+>
+> The sample bash commands can be used with any CI provider.
+> 
+> Baseline assumption: unit tests are able to generate a **coverage** folder when executed. This is the default folder that unit test frameworks use.
+
+
+
+The npm packages that will be used in this guide are:
+
+```bash
+npm install -D / yarn add --dev
+
+ngx-build-plus 
+
+istanbul-instrumenter-loader 
+
+@cypress/code-coverage 
+
+check-code-coverage
```

Comment:
All of them could become a simple list (without code indentation and without installation instruction) where every item links fo its reposiotry 😊